ros2lifecycle/test/test_cli.py

```diff
@@ -0,0 +1,342 @@
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes']) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['/lc_node'],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_list_all_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes', '-a']) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                '/_hidden_lc_node',
+                '/lc_node'
+            ],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_count_all_lifecycle_nodes(self):
+        with self.launch_lifecycle_command(arguments=['nodes', '-a', '-c']) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == launch_testing.asserts.EXIT_OK
+        output_lines = lifecycle_command.output.splitlines()
+        assert len(output_lines) == 1
+        assert int(output_lines[0]) == 2
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_set_lifecycle_node_invalid_transition(self):
+        with self.launch_lifecycle_command(
+            arguments=['set', '/lc_node', 'noop']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                'Unknown transition requested, available ones are:',
+                '- configure [1]',
+                '- shutdown [5]'
+            ],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_set_nonexistent_node_state(self):
+        with self.launch_lifecycle_command(
+            arguments=['set', '/nonexistent_node', 'configure']
+        ) as lifecycle_command:
+            assert lifecycle_command.wait_for_shutdown(timeout=10)
+        assert lifecycle_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Node not found'],
+            text=lifecycle_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_get_nonexistent_node_state(self):
```

Comment:
There should be a get without the hidden flag checking for a hidden node test

Reply:
Also, there's no `get` test valid. It's coupled with the `test_node_lifecycle`

Reply:
Added in https://github.com/ros2/ros2cli/pull/391/commits/fc5977ccf05dac4a6800cda1175f95335b249ebb. It's a bit annoying to not have proper fixture scoping.